### PR TITLE
multiple-pipeline-capture: simplify func_error_exit()

### PR DESCRIPTION
**Pull request by marc-hb**
As suggested by Chao in #521

Also remove no-op exit $? on the last line.

Signed-off-by: Marc Herbert <marc.herbert@intel.com>

### CODE REVIEW

**Comment by marc-hb** [comment]
https://sof-ci.01.org/softestpr/PR523/build407/devicetest/ had:
- jf-icl-rvp-hda-2/boot_failed
- One `aplay: pcm_write:2061: write error: Input/output error` on one platform in a different test.

None can possibly come from this PR.

**Comment by marc-hb** [comment]
No new shellcheck warning in https://travis-ci.org/github/thesofproject/sof-test/jobs/744366328

**Review by aiChaoSONG — DISMISSED** [review]
LGTM

**Comment by marc-hb** [comment]
Let's not merge this yet because it diverges `multiple-pipeline-capture` from `multiple-pipeline-playback` again while #525 tries to make them identical again.

**Comment by aiChaoSONG** [comment]
@marc-hb I think we have to merge this and #525 to make the two multiple-pipeline-* 99% identical, or the code of `func_error_exit()` diverges in two scripts.

**Comment by aiChaoSONG** [comment]
Merge this to make multiple-pipeline-playback.sh and multiple-pipeline-capture.sh 99.9% identical. Then we can merge the scripts into one.

**Comment by aiChaoSONG** [comment]
SOFCI TEST

**Comment by marc-hb** [comment]
No new warning in https://travis-ci.org/github/thesofproject/sof-test/jobs/744366328

https://sof-ci.01.org/softestpr/PR523/build422/devicetest is all green.

**Comment by marc-hb** [comment]
> Merge this to make multiple-pipeline-playback.sh and multiple-pipeline-capture.sh 99.9% identical. 

In "synchronization" commit  310fed5a13e I made the same change in `multiple-pipeline-playback.sh` before this #523 was merged. This made `multiple-pipeline-playback.sh` ahead of `multiple-pipeline-capture.sh`, sorry for the confusion.

**Comment by marc-hb** [comment]
The only failure in https://sof-ci.01.org/softestpr/PR523/build424/devicetest/?model=BSW_CYN_MAX98090&testcase=multiple-pipeline-capture looks like unrelated https://github.com/thesofproject/sof/issues/3462

**Review by aiChaoSONG — APPROVED** [review]
LGTM

**Comment by marc-hb** [comment]
Small, approved change which has been in review for 6 days, merging.

**Comment by marc-hb** [comment]
Now we need #531, new divergence (accidentally?) introduced by @xiulipan  in #529

**Comment by marc-hb** [comment]
> Now we need #531, new divergence (accidentally?) introduced by @xiulipan in #529

Thanks @xiulipan for the quick #531 review. Deduplication submitted in #532, not ready for merge today considering the other changes already merged but 100% ready for review, thanks!

**Pull request merged**
